packages/astro/src/core/build/plugins/plugin-pages.ts

```diff
@@ -25,8 +26,12 @@ function vitePluginPages(opts: StaticBuildOptions, internals: BuildInternals): V
 		async load(id) {
 			if (id === resolvedPagesVirtualModuleId) {
 				let importMap = '';
-				let imports = [];
+				const imports = [];
+				const exports = [];
+				const content = [];
```

Comment:
nit: I think we can type this as `string[]` since it can't be inferred by default.
```suggestion
				const imports: string[] = [];
				const exports: string[] = [];
				const content: string[] = [];
```